### PR TITLE
add a resource getter setter to newServiceHandler

### PR DESCRIPTION
**Pull request by ryanfkeepers**
This sets us up to remove the service declaration
from the two-step repo creation + init/config, and allows sdk consumers to return to ignoring service until acting on the operation.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :broom: Tech Debt/Cleanup

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

**Review by ashmrtn — COMMENTED** [review]
couple of high-level questions just so I understand the scope of the change and where we want to take it in the future since I'm not as familiar with this code (numbered for ease of reference):
1. this doesn't change the initialization for non-export operations correct? Right now `NewServiceHandler` is only for export so any changes to overall initialization could cause backup or restore issues
2. right now it seems like we're still planning on storing this resource getter in the controller. Is that the best place for it long-term? How much of the stuff in the controller can we pull out and place in the service handlers themselves as we update the architecture?

**Comment by ryanfkeepers** [comment]
> couple of high-level questions 

1. Correct.  That coverage will only happen as serviceHandler spreads throughout operations usage.
2. Could we move the code out of controller?  Sure, that seems fine.  Not particularly front of mind for me until we get the handlers fully implemented.

**Comment by ashmrtn** [comment]
just noting that this can be rebased for final review now (sorry for delay in getting the base PR merged)

**Comment by ryanfkeepers** [comment]
> just noting that this can be rebased for final review now (sorry for delay in getting the base PR merged)

No worries, and thanks for the heads up!

**Comment by ryanfkeepers** [comment]
@ashmrtn ready for final review

File: src/internal/m365/export.go [hunk 1]
```diff
@@ -17,6 +17,8 @@ func (ctrl *Controller) NewServiceHandler(
 	opts control.Options,
 	service path.ServiceType,
 ) (inject.ServiceHandler, error) {
+	ctrl.setResourceHandler(service)
```
Comment (ashmrtn) [inline src/internal/m365/export.go:20]: nit: just for now can you add a comment that this is more or less a noop? Export isn't using the resource client at all. Restore and backup will, but we haven't pushed the changes through that far yet

**Review by ashmrtn — APPROVED** [review]

File: src/internal/m365/export.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/internal/m365/export.go:20]: certainly can

**Comment by ryanfkeepers** [comment]
@ashmrtn something went wonky with the commit history in this branch.  I'm swapping it out with https://github.com/alcionai/corso/pull/4666

**Pull request closed without merging**